lib/main.js

```diff
@@ -3,96 +3,174 @@
-            buildServerUri: 'http://127.0.0.1:7100',
-        }, options);
+    constructor(options = {}) {
+        this.options = Object.assign(
+            {
+                buildServerUri: 'http://127.0.0.1:7100',
+            },
+            options
+        );
 
         this.transforms = [];
         this.plugins = [];
     }
 
-    transform (transform, options) {
+    transform(transform, options) {
         this.transforms.push({
             transform,
             options,
         });
     }
 
-    plugin (plugin, options) {
+    plugin(plugin, options) {
         this.plugins.push({
             plugin,
             options,
         });
     }
 
+    uploadJsFeed(files, options) {
+        const writer = new JsWriter(files, options, false, true);
 
-    /**
-     * Upload asset feed to asset server
-     */
+        this.transforms.forEach(entry => {
+            writer.transform(entry.transform, entry.options);
+        });
 
-    uploadFeed (files = [], options = {}) {
-        return new Promise((resolve, reject) => {
-            const writer = new Writer(files, options, false, true);
-
-            this.transforms.forEach((entry) => {
-                writer.transform(entry.transform, entry.options);
-            });
-
-            this.plugins.forEach((entry) => {
-                writer.plugin(entry.plugin, entry.options);
-            });
-
-            writer
-                .bundle()
-                .pipe(JSONStream.stringify())
-                .pipe(request.post({
-                    url: url.resolve(this.options.buildServerUri, 'feed'),
-                    json: true,
-                }, (error, response, body) => {
-                    if (error) {
-                        return reject(error);
-                    }
-                    if (response.statusCode === 200) {
-                        return resolve(body);
-                    }
-                    if (response.statusCode === 400) {
-                        return reject(new Error(body.message));
-                    }
-                    reject(new Error(`Asset build server responded with unknown error. Http status ${response.statusCode}`));
-                }));
+        this.plugins.forEach(entry => {
+            writer.plugin(entry.plugin, entry.options);
+        });
+
+        return post({
+            url: url.resolve(this.options.buildServerUri, endpoints.UPLOAD),
+            body: writer.bundle().pipe(JSONStream.stringify()),
         });
     }
 
+    uploadCssFeed(files) {
+        const writer = new CssWriter(files);
+        return post({
+            url: url.resolve(this.options.buildServerUri, endpoints.UPLOAD),
+            body: writer.pipe(JSONStream.stringify()),
+        });
+    }
 
     /**
-     * Make a bundle out of asset feeds on the asset server
+     * Upload asset feed to asset server
      */
 
-    createRemoteBundle (sources) {
-        return new Promise((resolve, reject) => {
```

Comment:
Are you sure about removing the promise here? You want the asserts to throw and not Promise.reject()?

Reply:
My feeling is that run time assert errors are usage errors for the developer trying to use the api and should explode spectacularly in their face 💥 💣 💥 💣  giving them concise useful information about what they did wrong. What are your thoughts on this?

Reply:
Indeed, as long as these errors occur during startup I guess its best to throw and stop.

Reply:
A function returning a promise should just reject, never throw.

Reply:
making the function `async` would solve it. not sure how we feel about node 6 support here?

Reply:
node 8 must be target IMO, at least until we know of someone who wants/needs node 6.

Reply:
hmm.

Alright yea I can buy that now that I think about it. As discussed yesterday async/await is nice way to not have to handle the 2 cases differently (other than declaring the function async)

The css writer and reader is node 8 only (that was the direction that PR discussion lead us in) so i'd be keen to do the same here also. I'm going to change to async and ask for forgiveness later if it turns out to be the wrong call ;)

Reply:
👍